contracts/src/Teleporter/upgrades/TeleporterUpgradeable.sol

```diff
@@ -60,7 +69,13 @@ abstract contract TeleporterUpgradeable is Context, ITeleporterReceiver, Reentra
      * @dev Initializes the {TeleporterUpgradeable} contract by getting `teleporterRegistry`
      * instance and setting `_minTeleporterVersion`.
      */
-    constructor(address teleporterRegistryAddress) {
+    // solhint-disable ordering
+    // solhint-disable-next-line func-name-mixedcase
+    function __TeleporterUpgradeable_init(address teleporterRegistryAddress)
+        internal
+        onlyInitializing
+    {
+        __ReentrancyGuard_init();
```

Comment:
Just for my own understanding, do you know why `ContextUpgradeable` doesn't have a `__ContextUpgradeable_init` function to be called here?

Reply:
`ContextUPgradeable` has no state to initialize with so the init functions are empty
```
    function __Context_init() internal onlyInitializing {
    }

    function __Context_init_unchained() internal onlyInitializing {
    }
```

Reply:
Gotcha, should we include calls to that here for completeness?  In the event that state is added to those contracts in a future version I image we'd want to call it.

Reply:
still need to follow up this. If you include the Context init call, for new versions we'd need to check through the inheritance hierarchy to make sure there's no duplicates of initialize calls anyways. Likely if called, we can call the `init_unchained` call. Also was looking at OZ's [OwnableUpgradeable](https://github.com/OpenZeppelin/openzeppelin-contracts-upgradeable/blob/master/contracts/access/OwnableUpgradeable.sol#L51) which also inherits `ContextUpgradeable` but they do not call the Context init function, so might ask OZ about that.

Reply:
following convention in https://github.com/ava-labs/teleporter/pull/397#discussion_r1684459392 to call `init` instead of `unchained` when not needed, went forward to call `Context_init` as well for completeness, and no duplicate initializations.

Reply:
Related OZ issue with their upgradeable contract transpiler not always calling the `init` functions of parent contracts https://github.com/OpenZeppelin/openzeppelin-contracts/issues/4690